parsely/parselyandroid/ParselyTracker.java

```diff
@@ -140,43 +289,26 @@ public void flush() {
     *
     *   @param queue The list of event dictionaries to serialize
     */
-    private void sendBatchRequest(ArrayList<Map<String, Object>> queue){
-        PLog("Sending batched request of size %d", queue.size());
-        if (queue == null || queue.size() == 0) {
+    private void sendBatchRequest(ArrayList<Map<String, Object>> events){
+        PLog("Sending batched request of %d events", events.size());
+        if (events == null || events.size() == 0) {
             return;
         }
-        Map<String, Object> batchMap = new HashMap<>();
-
-        // the object contains only one copy of the queue's invariant data
-        Map<String, String> pixelData = new HashMap<>();
-        pixelData = (Map<String, String>) queue.get(0).get("data");
-        if (!pixelData.get("parsely_site_uuid").equals(this.adKey) && this.adKey != null) {
-            pixelData.put("parsely_site_uuid", this.adKey);
-        }
-        batchMap.put("data", pixelData);
-        ArrayList<Map<String, Object>> events = new ArrayList<>();
-
-        for(Map<String, Object> event : queue){
-            String field = null, value = null;
-            if(event.get("url") != null){
-                field = "url";
-                value = (String)event.get("url");
-            } else if(event.get("postid") != null){
-                field = "postid";
-                value = (String)event.get("postid");
-            }
 
-            Map<String, Object> _toAdd = new HashMap<>();
-            _toAdd.put(field, value);
-            _toAdd.put("ts", String.format("%f", (double)event.get("ts")));
-            events.add(_toAdd);
-        }
+        // Put in a Map for the proxy server
+        Map<String, Object> batchMap = new HashMap<>();
         batchMap.put("events", events);
 
         PLog("Setting API connection");
-        new ParselyAPIConnection().execute(this.rootUrl + "mobileproxy", this.JsonEncode(batchMap));
-        PLog("Requested %s", this.rootUrl);
-        PLog("Data %s", this.JsonEncode(batchMap));
+        if (this.isDebug == true) {
+            PLog("Debug mode on. Not sending to Parse.ly");
+            this.eventQueue.clear();
+            this.purgeStoredQueue();
```

Comment:
Yeah, that's how it ended up here. I didn't want to muck around with the other, so I continued the tradition. Seemed acceptable tech debt at the time, especially since we're talking about a debug mode.